packages/tpetra/core/test/MultiVector/MultiVector_MicroBenchmark.cpp

```diff
@@ -99,26 +99,29 @@ namespace { // (anonymous)
       ::Tpetra::Details::ProfilingRegion region ("Tpetra Update Loop");
       for(size_t i=0; i<num_repeats; i++) {
         X.update(ONE,Y,ONE);
+        Kokkos::fence();
```

Comment:
You would want the fence on the outside of the for loop, but before ProfilingRegion closes.

Reply:
Also, as you pointed out today, Kokkos Profiling automatically fences at `popRegion()`.